### PR TITLE
Create TDD Test

### PR DESCRIPTION
**Pull request by FuadNabi**
PullRequestGIF

# In This Pull Request: 🎉🚀💫

### - Create a method called fizzbuzz that takes one argument, an integer N, and returns a string. The returned string is constructed following these rules:

- When N is divisible by 3, return "fizz".
- When N is divisible by 5, return "buzz".
- When N is divisible by 3 and 5, return "fizzbuzz".

### CODE REVIEW

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
+require_relative '../solver'
+
+describe Solver do
+  context 'Factorial Function Unit Testing'
```
Comment (1GORDON) [inline spec/solver_spec.rb:4]: Great job team 🤝

- [ ] As stated in the project requirements, kindly add tests for; - - `Negative value which raises an exception as an expectation`, and `Special case - when input is 0, the result should equal 1`.

File: README.md [hunk 2]
```diff
@@ -110,12 +86,18 @@ To run the project, execute the following command:
...
 - LinkedIn: [yussuf-Bashir](https://www.linkedin.com/in/yussuf-bashir/)
 
+👤 **Fuad Nabiyev**
+
+- GitHub: [@githubhandle](https://github.com/FuadNabi)
+- Twitter: [@twitterhandle](https://twitter.com/FuadNebiyev2)
+- LinkedIn: [LinkedIn](https://www.linkedin.com/in/fuad-nabiyev/)
+
 
 <p align="right">(<a href="#readme-top">back to top</a>)</p>
```
Comment (1GORDON) [inline README.md:103]: Great job team 🤝

- [ ] [Optional] You may kindly fix the broken MIT license link to make your readme file more professional.

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @FuadNabi & @YussufOB ✋

Good job so far!

## Highlights!
- Followed Gitflow ✔️ 
- Good commit messages ✔️
- Linters are passing ✔️
- Descriptive Pull request ✔️
- Professional readme ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/solver_spec.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  it 'Should return 40,320' do
+    solver = Solver.new
+    result = solver.factorial(8)
+    expect(result).to eq(40_320)
+  end
+
+  it 'Negative number factorial should rasie an error' do
+    expect { solver.factorial(-1) }.to raise_error(Exception)
+  end
+end
```
Comment (julie-ify) [inline spec/solver_spec.rb]: - [ ] Kindly add a test for the special case scenario when the number is 0 - its result should be 1.

**Review by julie-ify — CHANGES_REQUESTED** [review]
Hi @FuadNabi and @YussufOB,

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mwanawabangona — APPROVED** [review]
Hi @FuadNabi  ,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations!  

H
## Highlights
*   No linter errors :heavy_check_mark:
*   Repo looks professional :briefcase:
*   Solid pull request scroll
*   Good quality commits
*   Good job implementing the requested changes!! :clap: :clap: :clap:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
